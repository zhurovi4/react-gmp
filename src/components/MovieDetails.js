import React from "react";
import PropTypes from "prop-types";
import styles from "./MovieDetails.module.css";

const MovieDetails = ({ movie }) => {
  const { imageUrl, name, releaseYear, genres, rating, duration, description } =
    movie;
  const genresString = genres.join(", ");

  return (
    <div className={styles.movieInfo}>
      <img src={imageUrl} alt={name} className={styles.moviePoster} />
      <div>
        <div className={styles.movieTitle}>
          <h2>{name}</h2>
          <span className={styles.movieRating}>{rating}</span>
        </div>
        <div className={styles.movieGenre}>{genresString}</div>
        <div className={styles.movieDate}>
            <div>{releaseYear}</div>
            <div>{duration}</div>
        </div>
        <div className={styles.movieDescription}>{description}</div>
      </div>
    </div>
  );
};

MovieDetails.propTypes = {
  movie: PropTypes.shape({
    imageUrl: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    rating: PropTypes.number,
    duration: PropTypes.string,
    description: PropTypes.string.isRequired,
    releaseYear: PropTypes.number.isRequired,
    genres: PropTypes.arrayOf(PropTypes.string).isRequired,
  }).isRequired,
};

export default MovieDetails;
