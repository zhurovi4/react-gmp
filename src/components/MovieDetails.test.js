import React from "react";
import { render, screen } from "@testing-library/react";
import MovieDetails from "./MovieDetails";

describe("MovieDetails component", () => {
  const movie = {
    poster_path: "https://m.media-amazon.com/images/M/MV5BZmExNmEwYWItYmQzOS00YjA5LTk2MjktZjEyZDE1Y2QxNjA1XkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_QL75_UX285_CR0,2,285,422_.jpg",
    title: "A Good movie",
    vote_average: 4.5,
    duration: "1h 30m",
    overview: "This is an example movie.",
    release_date: '2004-11-11',
    genres: ["Action", "Comedy"],
  };

  it("renders movie details", () => {
    render(<MovieDetails movie={movie} />);
    expect(screen.getByAltText(movie.title)).toBeInTheDocument();
    expect(screen.getByText(movie.title)).toBeInTheDocument();
    expect(screen.getByText(movie.vote_average.toString())).toBeInTheDocument();
    expect(screen.getByText(movie.duration)).toBeInTheDocument();
    expect(screen.getByText(movie.overview)).toBeInTheDocument();
    expect(screen.getByText(movie.release_date.toString())).toBeInTheDocument();
    expect(screen.getByText(movie.genres.join(", "))).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = render(<MovieDetails movie={movie} />);
    expect(container).toMatchSnapshot();
  });
});
