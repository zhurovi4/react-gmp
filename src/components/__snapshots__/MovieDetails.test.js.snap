// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails component matches snapshot 1`] = `
<div>
  <div
    class="movieInfo"
  >
    <img
      alt="A Good movie"
      class="moviePoster"
      src="https://m.media-amazon.com/images/M/MV5BZmExNmEwYWItYmQzOS00YjA5LTk2MjktZjEyZDE1Y2QxNjA1XkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_QL75_UX285_CR0,2,285,422_.jpg"
    />
    <div>
      <div
        class="movieTitle"
      >
        <h2>
          A Good movie
        </h2>
        <span
          class="movieRating"
        >
          4.5
        </span>
      </div>
      <div
        class="movieGenre"
      >
        Action, Comedy
      </div>
      <div
        class="movieDate"
      >
        <div>
          2004-11-11
        </div>
        <div>
          1h 30m
        </div>
      </div>
      <div
        class="movieDescription"
      >
        This is an example movie.
      </div>
    </div>
  </div>
</div>
`;
